packages/flutter/lib/src/widgets/pinned_header_sliver.dart

```diff
@@ -0,0 +1,74 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'framework.dart';
+
+/// A sliver that keeps its Widget child at the top of the a [CustomScrollView].
+///
+/// This sliver is preferable to the general purpose [SliverPersistentHeader]
+/// for its relatively narrow use case because there's no need to create a
+/// [SliverPersistentHeaderDelegate] or to predict the header's size.
+///
+/// {@tool dartpad}
+/// This example demonstrates that the sliver's size can change. Pressing the
+/// floating action button replaces the one line of header text with two lines.
+///
+/// ** See code in examples/api/lib/widgets/sliver/pinned_header_sliver.0.dart **
+/// {@end-tool}
+class PinnedHeaderSliver extends SingleChildRenderObjectWidget {
```

Comment:
I could be wrong but we usually prefix with the `Sliver` moniker. :)

Reply:
You're right, that's a convention I should have followed. Will rename everything.